distributed/protocol.py

```diff
@@ -0,0 +1,113 @@
+"""
+The distributed message protocol consists of the following parts:
+
+1.  The length of the header, stored as a uint32
+2.  The header, stored as msgpack
+3.  The payload, stored as possibly compressed msgpack
+4.  A sentinel value
+
+**Header**
+
+The Header contains the following fields:
+
+* **decode**: boolean, optional
+    If true, then decode bytestrings in dict keys or the raw value
+    This is only necessary because msgpack treats all strings as bytes
+* **compression**: string, optional
+    One of the following: ``'snappy', 'lz4', 'zlib'`` or missing for None
+
+**Payload**
+
+The payload is any msgpack serializable value.  It may be compressed based
+on the header.
+
+**Sentinel**
+
+We often terminate each message with a sentinel value.  This happens
+outside of this module though and is not baked in.
```

Comment:
@quasiben @minrk @broxtronix if any of you have a moment I'd appreciate a quick sanity check on this start of a protocol for `distributed`.

I expect the header to grow more complex in the future (including versions, options to forward, and whatnot.)  For now though this is as simple as I can make it.  Am I missing anything critical here?